### PR TITLE
Slim sidebar interactions updates

### PR DESCRIPTION
**Pull request by stevedya**
Addresses #7918. Changes to the sidebar include

### Changes Made
- Removing the peeking attribute so the sidebar only opens when intentionally set to expanded mode by using expand or search or account functionalities
- Adding tooltips on link item hovers
- Expanding of slim sidebar when search is clicked and when account options are clicked

### Browsers Tested
Chrome 99 && Safari 15.2

### CODE REVIEW

File: client/src/components/Sidebar/menu/SubMenuItem.scss [hunk 1]
```diff
@@ -40,7 +45,7 @@
 
   > h2 {
     // w-min-h-[160px] and w-mt-[35px] classes are to vertically align the title and icon combination to the search input on the left
-    @apply w-min-h-[160px] w-mt-[35px] w-text-white w-mb-0 w-inline-flex w-flex-col w-justify-center w-items-center;
+    @apply w-min-h-[160px] w-mt-[35px] w-px-4 w-box-border w-text-center w-text-white w-mb-0 w-inline-flex w-flex-col w-justify-center w-items-center;
```
Comment (stevedya) [inline client/src/components/Sidebar/menu/SubMenuItem.scss:48]: Turns out we did need the text-center here for longer strings of text

File: client/scss/overrides/_vendor.tippy.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
+.tippy-box {
+  @apply w-bg-primary w-text-white;
+  font-size: 12px;
```
Comment (thibaudcolas) [inline client/scss/overrides/_vendor.tippy.scss:3]: Shouldn’t this come from our tokens, in rem, and be 14px at a minimum?

File: client/src/components/Sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -17,8 +17,11 @@ export interface ModuleRenderContext {
   key: number;
   slim: boolean;
   expandingOrCollapsing: boolean;
+  onAccountExpand: () => void;
+  onSearchClick: () => void;
   currentPath: string;
   strings: Strings;
+
```
Comment (thibaudcolas) [inline client/src/components/Sidebar/Sidebar.tsx:24]: Where do those extra blank lines come from? 😄

File: client/src/components/Sidebar/SidebarPanel.scss [hunk 4]
```diff
@@ -1,9 +1,10 @@
 .sidebar-panel {
+  @apply w-transition-all;
```
Comment (thibaudcolas) [inline client/src/components/Sidebar/SidebarPanel.scss:2]: I’m not too keen on setting transitions to "all" – could we be specific?

File: client/src/components/Sidebar/SidebarPanel.scss [hunk 5]
```diff
@@ -43,12 +35,18 @@
...
+  }
+
   @include media-breakpoint-up(sm) {
     @at-root .sidebar--slim #{&} {
       // Remove once we drop support for Safari 13.
       // stylelint-disable-next-line property-disallowed-list
-      left: calc(#{$menu-width-slim} - var(--width));
-      inset-inline-start: calc(#{$menu-width-slim} - var(--width));
+      left: #{$menu-width-slim};
+      inset-inline-start: #{$menu-width-slim};
```
Comment (thibaudcolas) [inline client/src/components/Sidebar/SidebarPanel.scss:49]:
```suggestion
      left: $menu-width-slim;
      inset-inline-start: $menu-width-slim;
```

File: client/src/components/Sidebar/modules/Search.scss [hunk 6]
```diff
@@ -30,7 +30,7 @@
     -webkit-font-smoothing: auto;
 
     .sidebar--slim & {
-      opacity: 0;
+      @apply hover:w-cursor-pointer w-opacity-0;
```
Comment (thibaudcolas) [inline client/src/components/Sidebar/modules/Search.scss:33]: Could we instead set `w-cursor-pointer` without `hover:`? Just so this is a bit shorter.

File: client/src/components/TippyWrapper/TippyWrapper.tsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import React, { useMemo } from 'react';
+import Tippy from '@tippyjs/react';
+
+// Conditionally render tippy tooltips surrounding elements
```
Comment (thibaudcolas) [inline client/src/components/TippyWrapper/TippyWrapper.tsx:4]: This needs to be a JSDoc-style `/** ` comment so it shows up in IDEs.

File: client/src/components/TippyWrapper/TippyWrapper.tsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+const TippyWrapper = ({ condition, children, label, placement }) =>
```
Comment (thibaudcolas) [inline client/src/components/TippyWrapper/TippyWrapper.tsx:5]: Could we think of a more descriptive name for this that doesn’t require knowing what Tippy is? `TooltipWrapper` perhaps?

File: package.json [hunk 9]
```diff
@@ -110,6 +111,7 @@
     "redux-thunk": "^2.3.0",
     "reselect": "^4.0.0",
     "telepath-unpack": "^0.0.3",
+    "tippy.js": "^6.3.7",
```
Comment (thibaudcolas) [inline package.json:114]: Do we actually need this in in addition to `@tippyjs/react`?

File: wagtail/admin/templates/wagtailadmin/admin_base.html [hunk 10]
```diff
@@ -4,6 +4,7 @@
 {% block css %}
     <link rel="stylesheet" href="{% versioned_static 'wagtailadmin/css/vendor/jquery-ui/jquery-ui-1.10.3.verdant.css' %}" />
     <link rel="stylesheet" href="{% versioned_static 'wagtailadmin/css/vendor/jquery.tagit.css' %}">
+    <link rel="stylesheet" href="{% versioned_static 'wagtailadmin/css/vendor/tippy.css' %}">
```
Comment (thibaudcolas) [inline wagtail/admin/templates/wagtailadmin/admin_base.html:7]: I don’t get what this points to – if we do need to load their styles, I’d prefer we do this in Sass via the main `core.css` file or `wagtailadmin.css` / similar "global CSS" files.

File: package.json (hunk 9 above)
Comment (stevedya) [reply package.json:114]: In the future (header updates) we will need this but at this time we might not :)

File: wagtail/admin/templates/wagtailadmin/admin_base.html (hunk 10 above)
Comment (stevedya) [reply wagtail/admin/templates/wagtailadmin/admin_base.html:7]: I might need your help with this one i struggled to path it properly 😣

**Review by thibaudcolas — CHANGES_REQUESTED** [review]
@stevedya the majority of the code changes look good to me. I don’t think we actually need the TooltipWrapper at all, would be nice to remove it and use `<Tippy>` directly if its `disabled` prop works as expected.

I spotted a few minor UI issues:

- The start of the bird logo animation looks off to me. I think it’s the lack of color transition in particular? The color change is very abrupt compared to the movement.
- On mobile, the same animation’s end state doesn’t look correct. The bird seems much lower  over its circle than on desktop
- When switching from slim to expanded mode, the "Search" icon moves around, even though its start and end position are the same. Could we keep it always fixed, like the other icons?
- Same for the account avatar, although it’s less distracting as it’s at the bottom so much further from the "toggle" control
- There is an 8px gap between the menu items and the sidebar footer. Could we remove it?
- On the slim version, the focus outline of the logo looks really odd
- Again on the slim version, it would be nice if the focus outline of the search menu item matched that of other menu items below
- And in the same area again in slim mode – after I tab to the search button, there is another tab stop that’s invisible. This element should either have a visible outline, or be skipped from the tab stops.
- The "Account" button’s focus outline should be inset (`show-focus-outline-inside` mixin), like other menu items, rather than the left and bottom sides being cropped

Here is a recap as a GIF with some of those issues:

interactions

There is a potentially bigger issue on Windows (could you make sure you always test there as part of making PRs in the future?), where scrollbars are making it impossible to see the arrows for menu items with sub-menus, and mess up with the tooltips’ positioning:

windows-sidebar-interactions

---

And finally – Is it intended that the "Account" menu item / avatar lacks a tooltip?

File: client/scss/settings/_variables.scss [hunk 11]
```diff
@@ -47,7 +47,7 @@ $color-green-dark: #157b57;
 $color-green-darker: #105d42; // White has AAA compatibility when this is used in bgd.
 $color-salmon: #f37e77;
 $color-salmon-light: #fcf2f2;
-$color-white: #fff;
+$color-white: #ffffff;
```
[the remark this answers is not in the document]
Comment (stevedya) [reply client/scss/settings/_variables.scss:50]: Thats so weird, i don't remember changing this one haha, yes i can definitely revert

File: client/src/components/TooltipWrapper/TooltipWrapper.tsx [hunk 12]
```diff
@@ -0,0 +1,32 @@
+import React, { useMemo } from 'react';
+import Tippy from '@tippyjs/react';
```
[the remark this answers is not in the document]
Comment (stevedya) [reply client/src/components/TooltipWrapper/TooltipWrapper.tsx:2]: Omg how did I not see this earlier, I spent too much time on the tippy wrapper 😅 . Thanks for finding this

**Comment by stevedya** [comment]
> @stevedya the majority of the code changes look good to me. I don’t think we actually need the TooltipWrapper at all, would be nice to remove it and use `<Tippy>` directly if its `disabled` prop works as expected.
> 
> I spotted a few minor UI issues:
> 
> - [x] * The start of the bird logo animation looks off to me. I think it’s the lack of color transition in particular? The color change is very abrupt compared to the movement.
> - [x] * On mobile, the same animation’s end state doesn’t look correct. The bird seems much lower  over its circle than on desktop
> - [x] * When switching from slim to expanded mode, the "Search" icon moves around, even though its start and end position are the same. Could we keep it always fixed, like the other icons?
> - [x] * Same for the account avatar, although it’s less distracting as it’s at the bottom so much further from the "toggle" control
> - [x] * There is an 8px gap between the menu items and the sidebar footer. Could we remove it?
> - [x] * On the slim version, the focus outline of the logo looks really odd
> - [x] * Again on the slim version, it would be nice if the focus outline of the search menu item matched that of other menu items below
> - [x] * And in the same area again in slim mode – after I tab to the search button, there is another tab stop that’s invisible. This element should either have a visible outline, or be skipped from the tab stops.
> - [x] * The "Account" button’s focus outline should be inset (`show-focus-outline-inside` mixin), like other menu items, rather than the left and bottom sides being cropped

> There is a potentially bigger issue on Windows (could you make sure you always test there as part of making PRs in the future?), where scrollbars are making it impossible to see the arrows for menu items with sub-menus, and mess up with the tooltips’ positioning:

> And finally – Is it intended that the "Account" menu item / avatar lacks a tooltip?

Hey @thibaudcolas I believe I have addressed and fixed all these issues now 😇. I was also able to style the scrollbar for windows and even on Firefox which works quite well I think. Here are some screenshots

**Windows Chrome 98**
Screen Shot 2022-03-27 at 12 20 58 PM

**Firefox 96 Windows** 
Screen Shot 2022-03-27 at 12 08 40 PM

**Review by thibaudcolas — APPROVED** [review]
Looking excellent @stevedya! I didn’t even know it was possible to do this kind of stuff with scrollbar styling.

**Comment by kaedroho** [comment]
Thanks @stevedya and @thibaudcolas this is looking much better now!

I've noticed a few quite minor issues on the current version:

 - The close animation for sub-menus doesn't seem to play when the menu is expanded
 - The account menu does a weird thing when you collapse the menu with the account menu open
 - The avatar suddenly jumps to the right when you collapse the menu
 - I wonder if the new `|->` icon should flip horizontally rather than roll? (or maybe not animate at all?) The rolling animation feels a little strange since the pipe is meant to represent the always vertical menu
 - The Bird seems to have two hover states (try slowly moving your mouse cursor from top to bottom and you'll see it's wing appears before the hover animation is triggered)

**Comment by thibaudcolas** [comment]
Thank you @kaedroho, I had spotted some of these but decided they weren’t big enough to warrant the risk of further conflicts 🙈  should have made it clearer in the PR itself. I have created #8311 to compile similar things we’d want to address.

**Pull request merged**